src/PSR7/Exception/Validation/InvalidBody.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenAPIValidation\PSR7\Exception\Validation;
+
+use OpenAPIValidation\PSR7\OperationAddress;
+use OpenAPIValidation\Schema\Exception\SchemaMismatch;
+use function sprintf;
+
+class InvalidBody extends AddressValidationFailed
+{
+    public static function becauseBodyDoesNotMatchSchema(
+        string $contentType,
+        OperationAddress $addr,
+        SchemaMismatch $prev
+    ) : self {
+        $exception          = static::fromAddrAndPrev($addr, $prev);
+        $exception->message = sprintf('Body does not match schema for content-type "%s" for %s', $contentType, $addr);
```

Comment:
I'm not sure for `SchemaMismatch` whether I should leave generic message and use `prev` for deep inspections or prepend prev message instead of generic text. I'll try both ways a bit later